code/modules/mob/living/emote.dm

```diff
@@ -113,22 +109,34 @@
 	key_third_person = "flaps"
 	message = "flaps their wings."
 	restraint_check = TRUE
+	var/wing_time = 20
 
 /datum/emote/living/flap/run_emote(mob/user, params)
 	. = ..()
 	if(ishuman(user))
 		var/mob/living/carbon/human/H = user
-		if((H.dna.features["wings"] != "None") && !("wings" in H.dna.species.mutant_bodyparts))
-			H.OpenWings()
-			addtimer(H, "CloseWings", 20)
+		var/open = FALSE
+		if(H.dna.features["wings"] != "None")
+			if("wingsopen" in H.dna.species.mutant_bodyparts)
+				open = TRUE
+				H.CloseWings()
+			else
+				H.OpenWings()
+			addtimer(H, "[open ? "Open" : "Close"]Wings", wing_time)
+
+/datum/emote/living/flap/aflap
+	key = "aflap"
+	key_third_person = "aflaps"
+	message = "flaps their wings ANGRILY!"
+	wing_time = 10
 
 /datum/emote/living/flip
 	key = "flip"
 	key_third_person = "flips"
 
 /datum/emote/living/flip/run_emote(mob/user, params)
 	. = ..()
-	if(!user.restrained() || !user.sleeping)
+	if(!user.sleeping)

```

Comment:
it won't work because how I made to return if there's no message(probably my own fault but I don't really know what to do there instead)